src/OnboardingSPA/components/Drawer/DrawerPanel/DesignColors.js

```diff
@@ -43,6 +78,58 @@ const DesignColors = () => {
 		updateThemeStatus,
 	} = useDispatch( nfdOnboardingStore );
 
+	useEffect( () => {
+		if ( THEME_STATUS_ACTIVE === themeStatus ) {
+			getColorStylesAndPatterns();
+			if ( currentData?.data?.colorStyle === 'custom' ) {
+				setIsAccordionClosed( false );
+				customColorsResetRef?.current?.scrollIntoView( {
+					behavior: 'smooth',
+					block: 'end',
+				} );
+			}
+		}
+	}, [ themeStatus ] );
+
+	/**
+	 * Fetches the color palette for the Drawer and sets the state
+	 * Contains: 'tailored' and 'custom-picker-grouping'
+	 */
+	const getColorStylesAndPatterns = async () => {
+		const globalStyles = await getGlobalStyles();
+		const colorPaletteResponse = await getThemeColors();
+		if ( colorPaletteResponse?.error ) {
+			return updateThemeStatus( THEME_STATUS_INIT );
+		}
+		if ( globalStyles?.error ) {
+			return updateThemeStatus( THEME_STATUS_INIT );
+		}
+		setColorPalettes( colorPaletteResponse?.body.tailored );
+		setCustomColorsMap(
+			colorPaletteResponse?.body[ 'custom-picker-grouping' ]
+		);
+		let selectedColorPalette;
+		let selectedColorsLocalTemp;
+		if ( ! ( currentData?.data?.colorStyle === '' ) ) {
+			selectedColorPalette =
+				globalStyles.body[ 0 ]?.settings?.color?.palette;
+			selectedColorsLocalTemp = stateToLocal( selectedColorPalette );
+			if ( currentData?.data?.colorStyle === 'custom' ) {
+				setCustomColors( selectedColorsLocalTemp );
+			}
+			setSelectedColors( selectedColorPalette );
+		} else {
+			setToDefaultPalette();
+		}
+	};
+
+	/**
+	 * Helper Function for state in global styles to local mapping
+	 *
+	 * @param {Array} selectedColorPalette - Array of Map structure similar to the one in Global Styles
+	 *                                     e.g. [{color: '#ffffff', name: 'Base', slug: 'base'}, ...]
+	 *                                     return -> {base: "#ffffff", contrast: "#404040", ... }
+	 */
 	function stateToLocal( selectedColorPalette ) {
```

Comment:
`stateToLocal` will work only with selected color palettes? If not, does it make sense to move to a utils file? Should a conversion function modify state?

Reply:
As this only has a use case in the colors file and it will only work with a color palette as almost all the states are different on almost every page.